cpu/nrf51/radio/nrf51_ble/nrf51_ble.c

```diff
@@ -0,0 +1,630 @@
+
+    /* switch int RX mode */
+    NRF_RADIO->TASKS_RXEN = 1;
+    _state = STATE_RX;
+}
+
+/* Get the BLE Access Address of the interface */
+int _get_access_address(uint8_t *val, size_t max_len)
+{
+    /* check parameters */
+    if (max_len < BLE_ACCESS_ADDR_LEN) {
+        return -EOVERFLOW;
+    }
+
+    /* get address */
+    val[0] = (uint8_t)(_access_addr >> 24);
+    val[1] = (uint8_t)(_access_addr >> 16);
+    val[2] = (uint8_t)(_access_addr >> 8);
+    val[3] = (uint8_t)(_access_addr);
+    return BLE_ACCESS_ADDR_LEN;
+}
+
+/* Set the BLE Access Address of the interface */
+int _set_access_address(uint8_t *val, size_t len)
+{
+    int is_rx = 0;
+
+    /* check parameters */
+    if (len != BLE_ACCESS_ADDR_LEN) {
+        return -EINVAL;
+    }
+
+    /* keep track of state */
+    while (_state == STATE_TX);
+
+    if (_state == STATE_RX) {
+        is_rx = 1;
+        _switch_to_idle();
+    }
+
+    /* set BLE access address */
+    _access_addr = (((uint32_t)val[0]) << 24) | (((uint32_t)val[1]) << 16) |
+                   (((uint32_t)val[2]) << 8) | val[3];
+
+    NRF_RADIO->PREFIX0 = ((_access_addr >> 24) & 0x000000FF);
+    NRF_RADIO->BASE0   = ((_access_addr << 8) & 0xFFFFFF00);
+
+    if (is_rx) {
+        _switch_to_rx();
+    }
+
+    return BLE_ACCESS_ADDR_LEN;
+}
+
+/* Get the BLE radio state */
+int _get_state(uint8_t *val, size_t max_len)
+{
+    ng_netopt_state_t state;
+
+    /* check parameters */
+    if (max_len < sizeof(ng_netopt_state_t)) {
+        return -EOVERFLOW;
+    }
+
+    switch (_state) {
+        case STATE_OFF:
+            state = NG_NETOPT_STATE_OFF;
+            break;
+
+        case STATE_IDLE:
+            state = NG_NETOPT_STATE_SLEEP;
+            break;
+
+        case STATE_RX:
+            state = NG_NETOPT_STATE_IDLE;
+            break;
+
+        case STATE_TX:
+            state = NG_NETOPT_STATE_TX;
+            break;
+
+        default:
+            return -ECANCELED;
+    }
+
+    memcpy(val, &state, sizeof(ng_netopt_state_t));
+    return sizeof(ng_netopt_state_t);
+}
+
+/* Set the BLE radio state */
+int _set_state(uint8_t *val, size_t len)
+{
+    ng_netopt_state_t state;
+
+    /* check parameters */
+    if (len != sizeof(ng_netopt_state_t)) {
+        return -EINVAL;
+    }
+
+    /* get target state */
+    memcpy(&state, val, len);
+
+    /* switch to target state */
+    switch (state) {
+        case NG_NETOPT_STATE_SLEEP:
+            _switch_to_idle();
+            break;
+
+        case NG_NETOPT_STATE_IDLE:
+            _switch_to_rx();
+            break;
+
+        default:
+            return -ENOTSUP;
+    }
+
+    return sizeof(ng_netopt_state_t);
+}
+
+/* Get the current BLE channel number of the interface */
+int _get_channel(uint8_t *val, size_t max_len)
+{
+    /* check parameters */
+    if (max_len < 2) {
+        return -EOVERFLOW;
+    }
+
+    /* get channel */
+    val[0] = freq_to_chan[0x3f & NRF_RADIO->FREQUENCY];
+    val[1] = 0;
+    return 2;
```

Comment:
Why do you return the `2` here? I think it could be useful to add a small documentation even for the static functions (at least when you want to implement special return values)